src/devices_models/devices/common/range_constraint.jl

```diff
@@ -106,27 +106,27 @@ function device_semicontinuousrange(psi_container::PSIContainer,
     varbin = get_variable(psi_container, binvar_name)
     ub_name = _middle_rename(cons_name, "_", "ub")
     lb_name = _middle_rename(cons_name, "_", "lb")
+    names = collect(keys(range_data))
```

Comment:
This call is not necessary if we keep the names vector. Any reason to remove the names vector and make this a dict?